### DISCUSSION TITLE
Problem with `brew test-bot --only-formulae` on github-actions with platform macOS

### DISCUSSION THREAD

**Discussion by azatoth** [Homebrew/discussions#5045] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.3-52-g3e537a2
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3e537a25f242578e5a47a6f3e9a2f6bd22b733c5
Last commit: 3 hours ago
Core tap HEAD: 34706eeeaeadbd0c9f238a6850def0b6d2e61e4d
Core tap last commit: 4 weeks ago
Core tap JSON: 11 Jan 12:59 UTC
Core cask tap HEAD: a7efd5454fc938369e9c410d97650aa2681d8e01
Core cask tap last commit: 4 weeks ago
Core cask tap JSON: 11 Jan 12:59 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BOOTSNAP: set
HOMEBREW_CACHE: /Users/runner/Library/Caches/Homebrew
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_COLOR: set
HOMEBREW_CURL_PATH: /usr/bin/curl
HOMEBREW_DEVELOPER: set
HOMEBREW_FAIL_LOG_LINES: 150
HOMEBREW_GIT_EMAIL: 1589480+BrewTestBot@users.noreply.github.com
HOMEBREW_GIT_NAME: BrewTestBot
HOMEBREW_GIT_PATH: /usr/bin/git
HOMEBREW_LOGS: /Users/runner/work/homebrew-tools/homebrew-tools/logs
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_EMOJI: set
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_NO_INSTALL_CLEANUP: set
HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: quad-core 64-bit ivybridge
Clang: 14.0.0 build 1400
Git: 2.37.1 => /usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 12.7.2-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2 => /Applications/Xcode_14.2.app/Contents/Developer
```

### Output of `brew doctor`

```shell
All steps passed!
```

### Description of issue

Having created a tap per https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap and therein created a formula for a tool. The included github-actions workflow test, while modified from the original, succeeds on ubuntu but fails on macOS due to it can't find git-lfs:

```
 ==> FAILED
Full fetch myorg/tools/tools --build-bottle output
  ==> Fetching tools from myorg/tools
  ==> Cloning https://github.com/myorg/my-source-repo.git
  Cloning into '/Users/runner/Library/Caches/Homebrew/tools--git'...
  git-lfs filter-process: git-lfs: command not found
  fatal: the remote end hung up unexpectedly
  warning: Clone succeeded, but checkout failed.
  You can inspect what was checked out with 'git status'
  and retry with 'git restore --source=HEAD :/'
```

The original template did specify macos-13, but as that one failed I tested with macos-12 which per https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources is the stable one.

I did in a previous step check for git and git-lfs and it for macos returned:
```
/usr/local/bin/git
/usr/local/bin/git-lfs
```

Thus I'm a bit in an impasse and any help or ideas will be welcome.

Regards;

**Comment by SMillerDev** [comment]
Yeah, I don't think anyone ever tried this with git-lfs. We'd welcome pull requests that helped, but it's not normally supported.

**Reply by azatoth** [reply]
But why does it fail on macos but succeeds on ubuntu? I have problem making sense of that.

**Reply by SMillerDev** [reply]
Support for filtering is just a lot more robust on macOS than Linux for Homebrew

**Reply by azatoth** [reply]
We were able to remove LFS from the repo in question and now the build succeeds on macos as well. 
I sadly don't understand how having more robust filtering support will in turn trigger this issue but I appreciate the response.

**Reply by SMillerDev** [reply]
Homebrew tries very hard to filter out any tools not directly related to the build, and I guess it's succeeding a little better on macOS than Linux since git-lfs is not a tool I think it considers a requirement
